Mathlib/Algebra/Category/ContinuousCohomology/Basic.lean

```diff
@@ -0,0 +1,159 @@
+
+/-- The functor taking a representation `rep` to the representation `C(G, rep)`. -/
+@[simps]
+def I : Action (TopModuleCat R) G ⥤ Action (TopModuleCat R) G where
+  obj := Iobj
+  map {M N} φ :=
+  { hom := TopModuleCat.ofHom (ContinuousLinearMap.compLeftContinuous _ _ φ.hom.hom)
+    comm g := by
+      ext f g'
+      show (M.ρ g ≫ φ.hom).hom (f (g⁻¹ * g')) = (φ.hom ≫ N.ρ g).hom (f (g⁻¹ * g'))
+      rw [φ.comm] }
+  map_id _ := rfl
+  map_comp _ _ := rfl
+
+instance : (I R G).Additive := ⟨by tauto⟩
+instance : (I R G).Linear R := ⟨by tauto⟩
+
+/-- The constant function `rep ⟶ C(G, rep)` as a natural transformation. -/
+@[simps]
+def const : 𝟭 _ ⟶ I R G where
+  app _ := { hom := TopModuleCat.ofHom (.const _ _), comm _ := rfl }
+  naturality _ _ _ := rfl
+
+namespace MultiInd
+
+/-- The n-th functor taking `M` to `C(G, C(G,...,C(G, M)))` (with n `G`s).
+These functors form a complex, see `MultiInd.complex`. -/
+def functor : ℕ → Action (TopModuleCat R) G ⥤ Action (TopModuleCat R) G
+  | 0     => 𝟭 _
+  | n + 1 => functor n ⋙ I R G
+
+/-- The differential map in `MultiInd.complex`. -/
+def d : ∀ n : ℕ, functor R G n ⟶ functor R G (n + 1)
+  | 0     => const R G
+  | n + 1 => whiskerLeft (functor R G (n + 1)) (const R G) - (by exact whiskerRight (d n) (I R G))
+
+lemma d_zero : d R G 0 = const R G := rfl
+
+lemma d_succ (n : ℕ) :
+    d R G (n + 1) = whiskerLeft (functor R G (n + 1)) (const R G) -
+      (by exact whiskerRight (d R G n) (I R G)) := rfl
+
+@[simp]
+lemma whiskerRight_zero {C D E : Type*} [Category C] [Category D] [Category E]
+    {F G : C ⥤ D} (H : D ⥤ E) [Limits.HasZeroMorphisms D] [Limits.HasZeroMorphisms E]
+    [H.PreservesZeroMorphisms] :
+    whiskerRight (0 : F ⟶ G) H = 0 := by aesop_cat
+
+@[reassoc (attr := simp)]
+lemma d_comp_d (n : ℕ) :
+    d R G n ≫ d R G (n + 1) = 0 := by
+  induction n with
+  | zero =>
+    rw [d_succ, Preadditive.comp_sub, sub_eq_zero]
+    rfl
+  | succ n ih =>
+    rw [d_succ R G (n + 1), Preadditive.comp_sub]
+    nth_rw 2 [d_succ]
+    rw [Preadditive.sub_comp, ← whiskerRight_comp, ih, whiskerRight_zero, sub_zero, sub_eq_zero]
+    rfl
+
+/-- The complex of functors taking `M` to `C(G, C(G,...,C(G, M)))`. -/
+def complex : CochainComplex (Action (TopModuleCat R) G ⥤ Action (TopModuleCat R) G) ℕ where
+  X := functor R G
+  d i j := if h : j = i + 1 then d R G i ≫ eqToHom (by subst h; rfl) else 0
+  d_comp_d' _ _ _ h₁ h₂ := by subst h₁ h₂; simp
+
+end MultiInd
+
+/-- The functor taking an `R`-linear `G`-representation `M` to complex of representations
+whose n-th component is `C(G, C(G,...,C(G, M)))` (with n `G`s).
+The `G`-invariant submodule of it is the homogeneous cochains. -/
+def multiInd : Action (TopModuleCat R) G ⥤ CochainComplex (Action (TopModuleCat R) G) ℕ where
+  obj M := (((evaluation _ _).obj M).mapHomologicalComplex _).obj (MultiInd.complex R G)
+  map {M N} f := (NatTrans.mapHomologicalComplex ((evaluation _ _).map f) _).app _
```

Comment:
This seems like a purely formal construction: given a cochain complex of functors C => D, you seem to get a a functor from C to the cochain complexes in D. Is it worth abstracting this?